include/libtorrent/storage.hpp

```diff
@@ -565,6 +565,8 @@ namespace libtorrent
 	{
 		virtual int file_op(int const file_index, std::int64_t const file_offset
 			, span<file::iovec_t const> bufs, storage_error& ec) = 0;
+
+		virtual ~fileop() {}
```

Comment:
the problem I have with these is that these types are never deleted "polymorphically", i.e. all concrete types are instantiated on the stack. just having this destructor here, I think, is confusing. It implies that it's used or necessary.